server/broadcast.go

```diff
@@ -236,7 +238,7 @@ func processSegment(cxn *rtmpConnection, seg *stream.HLSSegment) {
 		}
 		cond.L.Unlock()
 		if monitor.Enabled {
-			monitor.SegmentFullyTranscoded(nonce, seg.SeqNo, common.ProfilesNames(sess.Profiles), len(segHashes) == len(res.Segments))
+			monitor.SegmentFullyTranscoded(nonce, seg.SeqNo, common.ProfilesNames(sess.Profiles), len(segHashes) == len(res.Segments), errCode)
```

Comment:
Just noting that we'll exit early if there's an error with the download (in order to retry), so I hope we aren't using the SegmentFullyTranscoded call to wrap up any stateful tracking:  https://github.com/livepeer/go-livepeer/commit/2dcf826485969be9415cc0b93b2237d867ecc751

The params here are getting sort of long, would just passing in the error suffice to determine whether `len(segHashes) == len(res.Segments)` ? Does the `allSuccess` parameter convey any additional data?

Additionally, we might want to incorporate any errors from a failed sig check.

Reply:
> I hope we aren't using the SegmentFullyTranscoded call to wrap up any stateful tracking

We are. Adjust metrics to retries is my next task.

> would just passing in the error suffice to determine whether len(segHashes) == len(res.Segments)

Changed it to use `gotErr` flag.

> Additionally, we might want to incorporate any errors from a failed sig check.

Will do this later, when we will start doing with this information something else than just printing it to log.